controllers/users/checkAllowAnonDM.js

```diff
@@ -0,0 +1,49 @@
+const {User, UserFollowUser} = require('../../databases/models');
+const UsersFunction = require('../../databases/functions/users');
+const UserFollowUserFunction = require('../../databases/functions/userFollowUser');
+const Getstream = require('../../vendor/getstream');
+
+module.exports = async (req, res) => {
+  const {userId: self_user_id} = req;
+  let {post_id} = req.params;
+
+  const target_user_id = await Getstream.feed.getUserIdFromSource(res, 'post', {post_id});
+
+  let user = await User.findOne(
+    {where: {user_id: target_user_id}},
+    'user_id allow_anon_dm only_received_dm_from_user_following'
+  );
+
+  if (user) {
+    let following_status;
+    if (user?.only_received_dm_from_user_following) {
+      const signedUserId = await UsersFunction.findSignedUserId(User, target_user_id);
+      const currentUserStatus = await UserFollowUserFunction.checkTargetUserFollowStatus(
+        UserFollowUser,
+        self_user_id,
+        signedUserId
+      );
+
+      if (!currentUserStatus?.isTargetFollowingMe) {
+        following_status = false;
+      } else {
+        following_status = true;
+      }
+    }
+
+    return res.status(200).json({
+      status: 'success',
+      user: {
+        id: target_user_id,
+        allow_anon_dm: user.allow_anon_dm,
+        only_received_dm_from_user_following: user.only_received_dm_from_user_following,
+        following_status
+      }
+    });
+  } else {
+    return res.status(400).json({
+      status: 'error',
+      message: 'User Not Found'
+    });
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code looks good overall, but there's a potential issue with error handling. If the `Getstream.feed.getUserIdFromSource` function fails or returns an invalid user ID, the error is not properly handled and could lead to unexpected behavior.

Consider adding error handling around this function call to ensure that any errors are caught and handled appropriately. This will improve the robustness of your code.

```diff
+ try {
  const target_user_id = await Getstream.feed.getUserIdFromSource(res, 'post', {post_id});
+ } catch (error) {
+   return res.status(500).json({
+     status: 'error',
+     message: 'Failed to get user ID from source'
+   });
+ }
```

Also, in line 28 and 30, you're setting `following_status` to `false` or `true` based on the condition `!currentUserStatus?.isTargetFollowingMe`. You can simplify this by directly assigning the result of the condition to `following_status`.

```diff
- if (!currentUserStatus?.isTargetFollowingMe) {
-   following_status = false;
- } else {
-   following_status = true;
- }
+ following_status = currentUserStatus?.isTargetFollowingMe;
```
This simplification reduces the complexity of your code and makes it easier to read and understand.


<!-- This is an auto-generated comment by OSS CodeRabbit -->